Automattic/wp-calypso
# Signup: Use localStorage for adding shopping cart products

#### Changes proposed in this Pull Request

One of two remaining places in calypso which does not use the `@automattic/shopping-cart` package to interact with the `/me/shopping-cart` endpoint is signup. This PR modifies signup so that instead of saving cart items directly, it saves them to localStorage temporarily from which checkout will extract them when it loads (which is functionality that already exists for siteless carts).

Besides providing consistency with other cart management in calypso and separating the cart logic away from the signup infrastructure, this also allows us to remove `lib/signup/cart` which is a very old and poorly maintained cart management API that has caused us issues in the past.

Since localStorage is not available in some environments (eg: Safari private browsing mode), this also has a fallback singleton storage that should act like a session storage for the same purpose.

#### Testing instructions

- Visit `/start` and add a domain and a paid plan.
- Verify there are no errors.
- Verify that when you reach checkout, both the domain and the plan are in your cart.
